### ISSUE TITLE
evaluate.py reads a model output that does not exist, so every metric it reports is 0.0

### ISSUE THREAD

**Issue by engichang1467** [engichang1467/Open-Rig3R#47]
## Problem

`scripts/evaluate.py` reads a model output that the model has never produced, so every
metric it has ever reported is a placeholder value rather than a measurement.

`scripts/evaluate.py:82`:

```python
pred_pc = outputs.get('pointcloud_pred', torch.empty(0,3, device=device))
```

`pointcloud_pred` is not a key `Rig3R` emits. `RigAwareTransformerDecoder.forward` returns
`pointmap`, `pointmap_conf`, `pose_raymap`, `rig_raymap`, `camera_center_pose`,
`camera_center_rig`, and `features` — nothing else. The string appears exactly once in the
whole repository, at the line above:

```
$ grep -rn "pointcloud_pred" --include=*.py .
scripts/evaluate.py:82:        pred_pc = outputs.get('pointcloud_pred', torch.empty(0,3, device=device))
```

So `.get(...)` always falls through to its default and `pred_pc` is always an empty
`(0, 3)` tensor. Both metrics then take their empty-input early return:

```python
# utils/metrics.py
def chamfer_distance(pc1, pc2):
    if pc1.numel() == 0 or pc2.numel() == 0:
        return torch.tensor(0.0, device=pc1.device)

def rig_discovery_accuracy(pred_pc, gt_pc):
    if pred_pc.numel() == 0 or gt_pc.numel() == 0:
        return torch.tensor(0.0, device=pred_pc.device)
```

The script therefore prints, for any checkpoint and any dataset:

```
Average Chamfer Distance over N sequences: 0.000000
Average Rig Discovery Accuracy: 0.0000
```

A Chamfer distance of exactly zero reads as a perfect reconstruction. It means no
reconstruction happened at all.

## Second problem: the rig metric is being handed the wrong thing

`scripts/evaluate.py:85` passes a point cloud to a function that expects rig keypoints:

```python
rig_acc = rig_discovery_accuracy(pred_pc, gt_pc)
```

`rig_discovery_accuracy` Hungarian-matches predicted against ground-truth **camera mount
positions**, one per camera, with a 0.1 m threshold. `scripts/infer_rig_discovery.py:150`
uses it correctly, passing the translations recovered from the rig raymap:

```python
pred_rig_keypoints = torch.stack([p['t'] for p in recovered_poses])
r_acc = rig_discovery_accuracy(pred_rig_keypoints, gt_rig_keypoints)
```

Matching a dense scene point cloud against another dense point cloud is a different
quantity, and with `len(gt_pc)` in the denominator it is not bounded by 1. Fixing the
first problem without fixing this one would produce a number that is wrong in a subtler
way.

## Suggested fix

`scripts/infer_rig_discovery.py` already does all of this correctly and can be followed
directly:

1. **Build the point cloud from what the model actually predicts.** `reconstruct_pointcloud`
   takes the per-view pointmaps plus recovered poses:

   ```python
   recovered_poses = [recover_pose_closed_form(rig_raymap[b, n].reshape(H_p, W_p, 6))
                      for n in range(N)]
   pred_pc = reconstruct_pointcloud([pointmaps[b, n] for n in range(N)], recovered_poses)
   ```

2. **Rescale before any metric in metres.** Predictions are at `z̄`-normalized scale, so
   Chamfer distance and the 0.1 m threshold need `align_scale` from `utils/metrics.py`:

   ```python
   pred_pc = pred_pc * align_scale(pred_pc, gt_pc)
   ```

3. **Give `rig_discovery_accuracy` keypoints**, not a scene cloud — the recovered pose
   translations, scaled by the same factor.

4. **Consider reporting rig mAA too.** `rig_maa` is already imported and used by
   `infer_rig_discovery.py`; it is rotation-only and therefore scale-free.

At that point `evaluate.py` and `infer_rig_discovery.py` overlap almost entirely, so it may
be worth having one of them call into a shared helper rather than maintaining two copies of
the reconstruction path.

## Also worth tightening

`scripts/evaluate.py:36-38` hard-codes evaluation geometry:

```python
n_frames = 2
image_size = (128, 128)
batch_size = 1
```

with `patch_size=16` hard-coded again at line 58. None of these are read from the config,
so an evaluation silently mismatches a model trained at any other resolution or patch size
instead of failing loudly. `configs/evaluate.yaml` currently carries only `checkpoint` and
`data`.

## Scope

Independent of the loss and head work in #35, #37, #38, #39, #40 and #41 — this bug
predates all of them and is not made worse by any of them. Only item 2 above interacts
with #41, and `align_scale` already exists for it.

**Comment by engichang1467** [comment]
Since I've resolved #49, this issue is no longer relevant
